### ISSUE TITLE
Option 1  COLAB creation-  transformer training/retraining  (not LLM)

### ISSUE THREAD

**Issue by grewe** [grewe/InstructAware#4]
Create Colab for training a ideally pretrained transformer for retraining

**Comment by grewe** [comment]
Option 1-A :  we recreate the stucture of a transformer network given in some publication and train from scratch
(least desireable)


Option 1-B :   We use again someone elses architecture +  their PRE-TRAINED model on SOME dataset.   Goal is to find something if possible where the dataset has some kind of similarity to ours.  
(more desirable ---    lesser: dataset is not at all realted to ours    better: dataset is large, general    best: dataset is large and closer to our kind of dataset--instructional from input information)

**Comment by grewe** [comment]
FROM now on
Option 1 = Train from scratch simple transformer encoder-decoder

Option 2= Retrain text-to-text T5 (encoder-decoder)

Option 3= Retrain decoder OpenAI GPT

**Comment by grewe** [comment]
Create colab incorporated both training and testing/evaluation (**decide on metric to be used , use tensorboard or other additional training monitoring options).** Clearly specify training parameters used. Format data (see my suggestion for use of json --see colab at https://colab.research.google.com/drive/1DL-KhKorISuBOUxkRK-nXdV9Ae7g3DRb?usp=drive_link



**metrics used should be the same for all 3 options**


**metrics used should be the same for all 3 options**

**Comment by rakshitshah280701** [comment]
Option 1 - Training of Transformer from Scratch Colab file - 
https://colab.research.google.com/drive/1orAc8W9sziHX3ZqXsgQgrK4s7U0JCUiS?usp=drive_link

Option 1 - Traning of Transformer from Scratch Output File with Evaluation Metrics - https://drive.google.com/file/d/1IAJHweXDqQggQ1O7xA8h651ZsDclbwRZ/view?usp=drive_link

**Comment by rakshitshah280701** [comment]
Documentation added in Capstone Report doc
-Training
-Prediction/Testing
-Evaluation Metrics
-Graphs

**Comment by grewe** [comment]
1) quick retraining w/ fixed dataset
2) Waiting on documentation in Capston --must be narrative
